inc/Addon/Busting/FileBustingTrait.php

```diff
@@ -130,34 +128,20 @@ private function get_all_files() {
 			return false;
 		}
 
-		$dir = $this->filesystem->dirlist( $dir_path );
-
-		if ( false === $dir ) {
-			Logger::error(
-				'Could not get the directory contents.',
-				[
-					self::LOGGER_CONTEXT,
-					'path' => $dir_path,
-				]
-			);
-			return false;
-		}
-
-		if ( ! $dir ) {
-			return [];
-		}
+		$pattern = '/' . sprintf(
+			$this->escape_file_name( $this->filename_pattern ),
+			'([a-f0-9]{32}|local)'
+		) . '/';
```

Comment:
Question: When does the `$this->filename_pattern` change?

Why?

This could be an expensive process. I'm wondering if the pattern should run only when `$this->filename_pattern is set or changed.

What do you think?

Reply:
this is a trait so the class that use this trait has an attribute `filename_pattern` so for example 

https://github.com/wp-media/wp-rocket/blob/62462b4e8e34623444454a39ff588cd973a7a363/inc/Addon/GoogleTracking/GoogleAnalytics.php#L44

https://github.com/wp-media/wp-rocket/blob/3be25639a3c18b3058f29c7bda9c9ffbdb3c0f9e/inc/Addon/GoogleTracking/GoogleTagManager.php#L33

so it's constant for each class of them.
and keep in mind that this operation happens only one time due to this condition

https://github.com/wp-media/wp-rocket/blob/dd4c4052b6db0d91f9c07a6bfbe968203de6eb37/inc/Addon/Busting/FileBustingTrait.php#L87-L89

so what do you think?